Marlin/src/module/motion.cpp

```diff
@@ -792,15 +792,18 @@ void do_blocking_move_to_x(const_float_t rx, const_feedRate_t fr_mm_s/*=0.0*/) {
       fr_mm_s
     );
   }
-  void do_z_clearance(const_float_t zclear, const bool lower_allowed/*=false*/) {
-    if (DEBUGGING(LEVELING)) DEBUG_ECHOLNPGM("do_z_clearance(", zclear, ", ", lower_allowed, ")");
-    const float zdest = _MIN(zclear, Z_MAX_POS);
-    if (zdest == current_position.z || (!lower_allowed && zdest < current_position.z)) return;
+  void do_z_clearance(const_float_t zclear, const bool with_probe/*=true*/, const bool lower_allowed/*=false*/) {
+    UNUSED(with_probe);
+    float zdest = zclear;
+    TERN_(HAS_BED_PROBE, if (with_probe && probe.offset.z < 0) zdest -= probe.offset.z);
```

Comment:
This is changing the offset from Probe::do_z_raise HOTEND offset to probe.offset, is that going to cause any issues?
```
const float zoffs = DIFF_TERN(HAS_HOTEND_OFFSET, offset.z, hotend_offset[active_extruder].z);
```
There's also a now unused `UNUSED(with_probe);` two lines up.

Reply:
- I removed the `do_z_raise` method and just inlined the code, and it no longer refers to `probe.offset.z` there since it calls `do_z_clearance`.
- If `HAS_BED_PROBE` is not set then `with_probe` is stripped out in the precompiler by `TERN_`. Hence `UNUSED` is added to suppress the "unused" warning.

Reply:
So far in my testing there's nothing weird happening in `G28` or `G29`, though it does raise a little more than I would prefer. I've not tested `G29` with UBL, only ABL, but its clearance moves are very similar. I've also not tested `G35` yet, but that is next!

Reply:
> it does raise a little more than I would prefer.

Is that adding probe offset to clearance giving a higher Z?

Reply:
It is intended to consistently apply the behavior described in:
```cpp
/**
 * Z probes require clearance when deploying, stowing, and moving between
 * probe points to avoid hitting the bed and other hardware.
 * Servo-mounted probes require extra space for the arm to rotate.
 * Inductive probes need space to keep from triggering early.
 *
 * Use these settings to specify the distance (mm) to raise the probe (or
 * lower the bed). The values set here apply over and above any (negative)
 * probe Z Offset set with NOZZLE_TO_PROBE_OFFSET, M851, or the LCD.
 * Only integer values >= 1 are valid here.
 *
 * Example: `M851 Z-5` with a CLEARANCE of 4  =>  9mm from bed to nozzle.
 *     But: `M851 Z+1` with a CLEARANCE of 2  =>  2mm from bed to nozzle.
 */
```

Reply:
> Is that adding probe offset to clearance giving a higher Z?

Is that causing more of a raise than I would prefer? When the BLTouch probe is in HS mode it's not bad. And when it's not in HS mode, it's still not bad, since the probe will need to deploy fully at the next point. I was trying to get a tiny raise, just 1mm clearance for the tip of the deployed probe, but I (so far) didn't get it quite that low.

Reply:
Everything seems to be working well with this PR plus the (modified) changes that remain in your extra-clearance PR, but I will give it another round of testing later today. If you have a chance to try that combination also I would value your feedback. Pass it on!